lib/src/main/java/com/auth0/jwt/oicmsg/KeyBundle.java

```diff
@@ -0,0 +1,519 @@
+        } catch (Exception e) {
+            logger.error(e.getMessage());
+            throw new UpdateFailed("Couldn't make GET request to url: " + this.source);
+        }
+
+        if (statusCode == 304) {
+            this.timeOut = System.currentTimeMillis() + this.cacheTime;
+            this.lastUpdated = System.currentTimeMillis();
+
+            List<Key> keys = this.impJwks.get("keys");
+            if (keys != null) {
+                doKeys(keys);
+            } else {
+                logger.error("No 'keys' keyword in JWKS");
+                throw new UpdateFailed("No 'keys' keyword in JWKS");
+            }
+        } else if (statusCode == 200) {
+            this.timeOut = System.currentTimeMillis() + this.cacheTime;
+            try {
+                this.impJwks = parseRemoteResponse(response);
+            } catch (Exception exception) {
+                exception.printStackTrace();
+            }
+
+            if (!this.impJwks.keySet().contains("keys")) {
+                throw new UpdateFailed(this.source);
+            }
+
+            logger.debug("Loaded JWKS: " + response.toString() + " from " + this.source);
+            List<Key> keys = this.impJwks.get("keys");
+            if (keys != null) {
+                doKeys(keys);
+            } else {
+                logger.error("No 'keys' keyword in JWKS");
+                throw new UpdateFailed("No 'keys' keyword in JWKS");
+            }
+
+            Header[] headers = response.getHeaders("Etag");
+            if (headers != null) {
+                this.eTag = headers;
+            } else {
+                throw new KeyException("No 'Etag' keyword in headers");
+            }
+        } else {
+            throw new UpdateFailed("Source: " + this.source + " status code: " + statusCode);
+        }
+
+        this.lastUpdated = System.currentTimeMillis();
+        return true;
+    }
+
+    private JSONObject parseRemoteResponse(HttpResponse response) throws IOException, ParseException {
+        if (!response.getHeaders("Content-Type").equals("application/json")) {
+            logger.warn("Wrong Content_type");
+        }
+
+        logger.debug(String.format("Loaded JWKS: %s from %s", response.toString(), this.source));
+
+        return (JSONObject) new JSONParser().parse(EntityUtils.toString(response.getEntity()));
+    }
+
+    private boolean upToDate() {
+
+        boolean result = false;
+        if (!this.keys.isEmpty()) {
+            if (this.remote) {
+                if (System.currentTimeMillis() > this.timeOut) {
+                    if (update()) {
+                        result = true;
+                    }
+                }
+            }
+        } else if (this.remote) {
+            if (update()) {
+                result = true;
+            }
+        }
+
+        return result;
+    }
+
+    public boolean update() {
+        boolean result = true;
+        if (!this.source.isEmpty()) {
```

Comment:
```
if (this.source.isEmpty()) {
return true;
}

//rest of the code

```

Reply:
I found the current way it's written more natural to read.